zeppelin-interpreter/src/main/java/org/apache/zeppelin/scheduler/AbstractScheduler.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.scheduler;
+
+import org.apache.zeppelin.interpreter.InterpreterResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.LinkedBlockingQueue;
+
+/**
+ * Abstract class for scheduler implementation. Implementor just need to implement method
+ * runJobInScheduler.
+ */
+public abstract class AbstractScheduler implements Scheduler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AbstractScheduler.class);
+
+  protected String name;
+  protected volatile boolean terminate = false;
+  protected BlockingQueue<Job> queue = new LinkedBlockingQueue<>();
+  protected Map<String, Job> jobs = new ConcurrentHashMap<>();
+
+
+  public AbstractScheduler(String name) {
+    this.name = name;
+  }
+
+  public String getName() {
+    return this.name;
+  }
+
+  @Override
+  public List<Job> getAllJobs() {
+    return new ArrayList<>(jobs.values());
+  }
+
+  @Override
+  public Job getJob(String jobId) {
+    return jobs.get(jobId);
+  }
+
+  @Override
+  public void submit(Job job) {
+    job.setStatus(Job.Status.PENDING);
+    queue.add(job);
+    jobs.put(job.getId(), job);
+  }
+
+  @Override
+  public Job cancel(String jobId) {
+    Job job = jobs.remove(jobId);
+    job.abort();
+    return job;
+  }
+
+  @Override
+  public void run() {
+    while (!terminate) {
+      Job runningJob = null;
+      try {
+        runningJob = queue.take();
+      } catch (InterruptedException e) {
+        LOGGER.warn("FIFOScheduler is interrupted", e);
```

Comment:
It seems that the name of the scheduler needs to be corrected after cut+paste.
May be:
```java
LOGGER.warn("{} is interrupted", getClass().getSimpleName(), e);
```

Reply:
Thanks @aka-demik  I have fixed it